src/Assert.php

```diff
@@ -219,6 +219,8 @@ public static function string($value, $message = '')
     }
 
     /**
```

Comment:
I'd love to add these, however we ran into issue with these assertions as mentioned in 

#144 & #145.

Reply:
Not sure I follow: all type annotations I added are explicitly tested (besides `!empty`, which I'm gonna try while at the airport).


`oneOf` was skipped , since there's no way to infer a union/intersection type from an array of values right now.

Reply:
Reading #145: if any of the assertions result in a type restriction, then the addition of a type assertion is net downstream added value

Reply:
I think it is a good idea to increase the information available about the types, but then we run into issues like this: https://psalm.dev/r/db940cd402 as psalm then think we are doing redundant work. Because the type assertion isn't the main goal of the method, its a side-effect

Reply:
For me, the type assertion is the main goal of the method: the runtime exception is the side-effect (which is needed, but an unhappy path)

Reply:
Besides that, the example at https://psalm.dev/r/db940cd402 seems to be an upstream bug: `!empty` and `string` can be intersected, while the redundancy check seems to ignore that